sdk/resourcemanager/azure-resourcemanager-network/src/test/java/com/azure/resourcemanager/network/NetworkInterfaceOperationsTests.java

```diff
@@ -509,6 +511,266 @@ public void canAssociateNatGateway() {
         Assertions.assertEquals(gateway2.id(), subnet2.natGatewayId());
     }
 
+    @Test
+    public void canCreateIpConfigurationsWithDeleteOption() {
+        String subnetName = generateRandomResourceName("subnet-", 15);
+        resourceManager.resourceGroups().define(rgName).withRegion(Region.US_EAST).create();
+        Network vnet = networkManager.networks()
+            .define(generateRandomResourceName("vnet-", 15))
+            .withRegion(Region.US_EAST)
+            .withExistingResourceGroup(rgName)
+            .withAddressSpace("10.0.0.0/28")
+            .withSubnet(subnetName, "10.0.0.0/28")
+            .create();
+
+        NetworkInterface nic = networkManager.networkInterfaces()
+            .define(generateRandomResourceName("nic-", 15))
+            .withRegion(Region.US_EAST)
+            .withExistingResourceGroup(rgName)
+            .withExistingPrimaryNetwork(vnet)
+            .withSubnet(subnetName)
+            .withPrimaryPrivateIPAddressDynamic()
+            .withNewPrimaryPublicIPAddress()
```

Comment:
Can we do `withNewPrimaryPublicIPAddress(DeleteOptions)`?

Reply:
In `Create` mode is OK. What about `Update` mode?

Reply:
I mean in `WithPrimaryPublicIPAddress`, for both create and update stage.
OK, I saw there are a few overloads of this method, maybe we see if we need those later. You may keep it as it is.

Reply:
Nit, seems we can do `withNewPublicIpAddress(DeleteOptions)` in secondary ip configuration afterall:
```java
networkManager
  .networkInterfaces()
  .define(nicNames[2])
  .withRegion(Region.US_EAST)
  .withNewResourceGroup(rgName)
  .withExistingPrimaryNetwork(network)
  .withSubnet("subnet1")
  .withPrimaryPrivateIPAddressDynamic()
  .defineSecondaryIPConfiguration("nicip2")
    .withExistingNetwork(network)
    .withSubnet("subnet1")
    .withPrivateIpAddressDynamic()
    .withNewPublicIpAddress(DeleteOptions.DELETE) // this line
  .attach());
```
Changing this part:
https://github.com/Azure/azure-sdk-for-java/blob/714ba60364dd9c54362e100570f62786b59b4498/sdk/resourcemanager/azure-resourcemanager-network/src/main/java/com/azure/resourcemanager/network/models/NicIpConfiguration.java#L140-L142

into 
```java
interface WithPublicIPAddress<ParentT>
    extends HasPublicIpAddress.DefinitionStages.WithPublicIPAddress<WithAttach<ParentT>> {
    WithAttach<ParentT> withNewPublicIpAddress(DeleteOptions deleteOptions);
}
```
would do the trick.

Anyways, again, quite a few overloads.

Reply:
@XiaofeiCao 
Based on your ideas, I made a partial fix to the code. However, `NetworkInterface.withPrimaryPublicIPAddressDeleteOptions` and `NicIpConfiguration.withPublicIPAddressDeleteOptions` in update mode are retained for updating `DeleteOption` without redefining `IpConfiguration`.

Reply:
Thanks Hong. Above is just a proposal... Though after seeing the new interfaces, I do prefer them over the old ones.
So basically it's:
```java
// create
NetworkInterface nic = networkManager.networkInterfaces()
    .define(generateRandomResourceName("nic-", 15))
    .withRegion(Region.US_EAST)
    .withExistingResourceGroup(rgName)
    .withExistingPrimaryNetwork(vnet)
    .withSubnet(subnetName)
    .withPrimaryPrivateIPAddressDynamic()
    .withNewPrimaryPublicIPAddress(DeleteOptions.DELETE)
    .defineSecondaryIPConfiguration("secondary")
      .withExistingNetwork(vnet)
      .withSubnet(subnetName)
      .withPrivateIpAddressDynamic()
      .withNewPublicIpAddress(DeleteOptions.DETACH)
    .attach()
    .create();

// update
nic.update()
    .updateIPConfiguration("secondary")
      .withPublicIPAddressDeleteOptions(DeleteOptions.DELETE)
      .parent()
    .apply();

```
right?(though I don't see nic.update().withPrimaryPublicIpAddressDeleteOptions(DeleteOptions.DELETE)?)

Reply:
Regarding `nic.update().withPrimaryPublicIpAddressDeleteOptions(DeleteOptions)`, there is a test case following as : https://github.com/Azure/azure-sdk-for-java/pull/39096/files#diff-14abd6d0fae6e783bc0a658c1456c7f96f0c55674fa14ce8cb994cbde0ffe107R650

Reply:
Overall use cases look good to me. Though the overloads everywhere seem too much to me. Would you see if it's possible to extract these overloads and put them into a new interface, and let all extends this interface, like `HasPublicIpAddress`?

Reply:
Add interface `WithDefinitionSecondaryOrUpdatePublicIPAddress` into `NetworkInterface` in new version.